packages/@aws-cdk/aws-cloudfront/lib/web_distribution.ts

```diff
@@ -511,7 +517,30 @@ interface BehaviorWithOrigin extends Behavior {
  *
  *
  */
-export class CloudFrontWebDistribution extends cdk.Construct implements IDistribution {
+export class CloudFrontWebDistribution extends CloudFrontWebDistributionBase {
+  /**
+   * Creates a Web Distribution construct that represents an external bucket.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param distributionArn
+   */
+  public static fromWebDistributionArn(scope: cdk.Construct, id: string, distributionArn: string): IDistribution {
+    const stack = cdk.Stack.of(scope);
+    /*const region = stack.region;
+    const urlSuffix = stack.urlSuffix;*/
+
+    class Import extends CloudFrontWebDistributionBase {
+      public readonly distributionId = stack.parseArn(distributionArn).resourceName!;
+      // TODO
+      public readonly domainName = "";
```

Comment:
I'm not sure how to retrieve the `domainName` property, or any of the `CloudFrontWebDistributionProps`. 
Should they be manually by the user in their stack, or should I do something similar to `Vpc.fromLookup`?

Reply:
This implies that `domainName` does not belong to `IDistribution`

Reply:
Fair enough. So we `@deprecate` it before removing it in v2?

And concerning the other props, we just leave them empty? Won't that cause issues during deployment if another resource tries to access one of them?

Reply:
It’s an experimental package. You can break...